### PR TITLE
Functional Component Transformation (Part 2)

### PR DESCRIPTION
**Pull request by joe-bell**
## Description

Continuation of #341 

Further transformation associated to #259 

**Current progress**

- [x] `/components` – 32/32 (**+11**)
- [ ] `/jupyter` – 0/1 – _low priority_
- [ ] `/utils` – 4/5 – _`webserver-utils` low priority_
- [ ] `/views` – 18/19 (**+4**)
- [x] `App.jsx`
- [x] `main.jsx`

### Checklist

- [x] The documentation reflects the changes.
- [ ] I have manually tested the application to make sure the changes don’t cause any downstream issues, which includes making sure `./orchest status --ext` is not reporting failures when Orchest is running.
- [ ] In case I changed code in the `orchest-sdk`, I updated its version according to [SemVer](https://semver.org/) in its `_version.py` and updated the version compability table in its `README.md`

- [ ] In case I changed one of the services’ `models.py` I have performed the appropriate database migrations.

### CODE REVIEW

**Comment by joe-bell** [comment]
⚠️ UNTESTED - `ProjectsView`

File: services/orchest-webserver/client/src/views/FilePreviewView.jsx [hunk 1]
```diff
@@ -131,142 +105,144 @@ class FilePreviewView extends React.Component {
...
+            refManager.refs.htmlNotebookIframe.contentWindow.scrollTo(
+              refManager.refs.htmlNotebookIframe.contentWindow.scrollX,
               value
             );
           },
           () => {
-            return this.refManager.refs.htmlNotebookIframe.contentWindow
-              .scrollY;
+            return refManager.refs.htmlNotebookIframe.contentWindow.scrollY;
           },
```
Comment (joe-bell) [inline services/orchest-webserver/client/src/views/FilePreviewView.jsx:200]: @ricklamers when this PR is ready to review (after #341) I'll need to have a chat with you about the `scrollPosition` and `setWithRetry` logic – wasn't quite able to wrap my head around what's going on here and want to make sure no existing behaviour is compromised by the refactor

**Comment by joe-bell** [comment]
⚠️ **Todo**

- `EditJobView` - Move `loadView` functionality into a `useEffect` to prevent an `unsavedChanges` warning displaying

File: services/orchest-webserver/client/src/components/PipelineDetailsProperties.jsx [hunk 2]
```diff
@@ -51,7 +54,9 @@ class PipelineDetailsProperties extends React.Component {
 
   componentWillUnmount() {
     this.promiseManager.cancelCancelablePromises();
+    // @ts-ignore
```
Comment (joe-bell) [inline services/orchest-webserver/client/src/components/PipelineDetailsProperties.jsx:57]: Removing jQuery is out-of-scope for this task IMO, but it's in the plan for later

File: services/orchest-webserver/client/src/components/PipelineDetails.jsx [hunk 3]
```diff
@@ -1,229 +1,178 @@
...
-    });
-  }
+  React.useEffect(() => {
+    // overflow checks
+    overflowChecks();
 
-  getPaneWidth() {
-    let initialPaneWidth = 450;
-    let paneWidth = initialPaneWidth;
+    $(window).on("resize.pipelineDetails", overflowChecks.bind(this));
```
Comment (ricklamers) [inline services/orchest-webserver/client/src/components/PipelineDetails.jsx:69]: Why is this binding `this`?

File: services/orchest-webserver/client/src/components/ParamTree.jsx [hunk 4]
```diff
@@ -1,22 +1,30 @@
...
+ *  editParameter?: (parameterKey: any, key: any) => void;
+ *  strategyJSON: any;
+ * }} TParamTreeProps
+ *
+ * @type React.FC<TParamTreeProps>
+ */
+const ParamTree = (props) => {
+  const context = useOrchest();
+
+  console.log(props);
```
Comment (ricklamers) [inline services/orchest-webserver/client/src/components/ParamTree.jsx:25]: Probably want to get rid of this.

File: services/orchest-webserver/client/src/components/PipelineDetails.jsx (hunk 3 above)
Comment (joe-bell) [reply services/orchest-webserver/client/src/components/PipelineDetails.jsx:69]: Not entirely sure, I just copied what was there previously
https://github.com/orchest/orchest/pull/369/files/606a7c351741d85fc1d20055811b3a248898ce1e#diff-816a29372714b3120bb9102804a6ea480c994c338c617bc4738beea12421c94bL44

**Comment by joe-bell** [comment]
After co-ordinating with @fruttasecca, we think this is safe to merge

**Pull request merged**